File: Serenity.Script.UI/FilterPanel/FilteringTypes/IntegerFiltering.cs
using jQueryApi;
using Serenity.Data;
using System;
using System.Collections.Generic;

namespace Serenity
{
    public class IntegerFiltering : BaseEditorFiltering
    {
        public IntegerFiltering()
            : base(typeof(IntegerEditor))
        {
        }

        public override List<FilterOperator> GetOperators()
        {
            return AppendNullableOperators(
                AppendComparisonOperators(
                    new List<FilterOperator>()));
        }
    }
}

File: packages/corelib/src/ui/filtering/integerfiltering.ts
import { nsSerenity } from "../../base";
import { IntegerEditor } from "../editors/integereditor";
import { BaseEditorFiltering } from "./baseeditorfiltering";
import { FilterOperator } from "./filteroperator";

/**
 * Filtering handler for integer fields using an integer editor.
 */
export class IntegerFiltering extends BaseEditorFiltering<IntegerEditor> {
    static override[Symbol.typeInfo] = this.registerClass(nsSerenity);

    /**
     * Creates an integer filtering handler.
     */
    constructor() {
        super(IntegerEditor);
    }

    /**
     * Returns the operators supported by this filtering handler.
     * @returns The operators.
     */
    getOperators(): FilterOperator[] {
        return this.appendNullableOperators(this.appendComparisonOperators([]));
    }
}